vgridjs/src/timeline_track.tsx

```diff
@@ -295,29 +320,47 @@ interface TicksProps {
 }
 
 /** Ticks at the bottom of the timeline indicating video time at regular intervals */
-let Ticks: React.SFC<TicksProps> = observer((props) => {
-  let start = props.timeline_bounds.start;
-  let end = props.timeline_bounds.end;
-  let duration = end - start;
-  let ticks = _.range(start, end, duration / props.num_ticks);
-
-  return <svg className='timeline-ticks' style={{width: props.timeline_width, height: props.height}}>
-    {ticks.map((tick, i) => {
-       let hours = Math.floor(tick / 3600);
-       let minutes = Math.floor(60 * (tick / 3600 - hours));
-       let seconds = Math.floor(60 * (60 * (tick / 3600 - hours) - minutes));
-       let time_str = `${hours.toString().padStart(2, '0')}:${minutes.toString().padStart(2, '0')}:${seconds.toString().padStart(2, '0')}`;
-       let x = time_to_x(tick, props.timeline_bounds, props.timeline_width);
-
-       return <g key={i}>
-         <line
-           x1={x} x2={x} y1={0} y2={props.height / 2}
-           stroke="black" strokeWidth={1} />
-         <text x={x} y={props.height} textAnchor="middle">{time_str}</text>
-       </g>;
-    })}
-  </svg>;
-});
+@observer class Ticks extends React.Component<TicksProps, {}> {
+  private canvasRef : React.RefObject<HTMLCanvasElement>;
+  
+  constructor (props: TicksProps) {
+    super(props);
+    this.canvasRef = React.createRef();
+  }
+
+  componentDidMount() {
+    let start = this.props.timeline_bounds.start;
+    let end = this.props.timeline_bounds.end;
+    let duration = end - start;
+    let ticks = _.range(start, end, duration / this.props.num_ticks);
+    const canvas = this.canvasRef.current;
```

Comment:
What's the goal of turning this into a canvas? Seems like normal text elements should work just fine here.

Reply:
I believe that before the text was rendered using svg. Canvas might be a bit faster, but you bring up a good point that this rendering should be pretty fast either way due to the constant number of ticks. Will leave this issue open for now.